packages/astro/components/ViewTransitions.astro

```diff
@@ -4,9 +4,10 @@ type Fallback = 'none' | 'animate' | 'swap';
 export interface Props {
 	fallback?: Fallback;
 	handleForms?: boolean;
+	intraPageTransitions?: boolean;
```

Comment:
Previously, we had two passes through processing: navigation to different pages and a shortcut for navigation to locations on the same page. The routing processing is now unified, but it requires an explicit opt-in for view transitions to destinations on the same page.

Reply:
We currently do not do a VT at all on the same page. I'm thinking maybe that was a mistake. I think we should (in the future, not now) change it so that:

1. Hash clicks do not do VT.
2. Non-hash clicks to same page do.

This mirrors regular page navigation, if you click a link to the same page it does trigger page navigation, unless it's to a hash link.

Let me know if I am forgetting the reason we did this (probably).

Reply:
I remember we decided that we didn't need VT on "self links". 
With the `viewTransition` object now available via the `astro:before-swap` event, users can call `viewTransition.skipTransition()` to skip transitions. The default should indeed be: No VT to targets on the same page, unless the target has no hash. 

If we want to make this change, it will now be a one line change in `router:transform()`. Currently, I have left the semantics as they are so as not to introduce incompatibilities

Reply:
So the plan is for 4.0 to come out in ~1 month. Given that this would be released at earliest on 11/23, I'm thinking we shouldn't add this flag to only remove it in a couple of weeks. Instead we can keep intrapage navigation as it has always been and then in 4.0 default to allowing VT in 4.0.

Reply:
Fine with me!